pkg/loader/compose/v3.go

```diff
@@ -0,0 +1,236 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package compose
+
+import (
+	libcomposeyaml "github.com/docker/libcompose/yaml"
+	"io/ioutil"
+	"strings"
+
+	"k8s.io/kubernetes/pkg/api"
+
+	"github.com/docker/cli/cli/compose/loader"
+	"github.com/docker/cli/cli/compose/types"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/kubernetes-incubator/kompose/pkg/kobject"
+	"github.com/pkg/errors"
+)
+
+// The purpose of this is not to deploy, but to be able to parse
+// v3 of Docker Compose into a suitable format. In this case, whatever is returned
+// by docker/cli's ServiceConfig
+func parseV3(files []string) (kobject.KomposeObject, error) {
+
+	// In order to get V3 parsing to work, we have to go through some preliminary steps
+	// for us to hack up github.com/docker/cli in order to correctly convert to a kobject.KomposeObject
+
+	// Gather the working directory
+	workingDir, err := getComposeFileDir(files)
+	if err != nil {
+		return kobject.KomposeObject{}, err
+	}
+
+	// Load and then parse the YAML first!
+	loadedFile, err := ioutil.ReadFile(files[0])
+	if err != nil {
+		return kobject.KomposeObject{}, err
+	}
+
+	// Parse the Compose File
+	parsedComposeFile, err := loader.ParseYAML(loadedFile)
+	if err != nil {
+		return kobject.KomposeObject{}, err
+	}
+
+	// Config file
+	configFile := types.ConfigFile{
+		Filename: files[0],
+		Config:   parsedComposeFile,
+	}
+
+	// Config details
+	// Environment is nil as docker/cli loads the appropriate environmental values itself
+	configDetails := types.ConfigDetails{
+		WorkingDir:  workingDir,
+		ConfigFiles: []types.ConfigFile{configFile},
+		Environment: nil,
+	}
+
+	// Actual config
+	// We load it in order to retrieve the parsed output configuration!
+	// This will output a github.com/docker/cli ServiceConfig
+	// Which is similar to our version of ServiceConfig
+	config, err := loader.Load(configDetails)
+	if err != nil {
+		return kobject.KomposeObject{}, err
+	}
+
+	// TODO: Check all "unsupported" keys and output details
+	// Specifically, keys such as "volumes_from" are not supported in V3.
+
+	// Finally, we convert the object from docker/cli's ServiceConfig to our appropriate one
+	komposeObject, err := dockerComposeToKomposeMapping(config)
+	if err != nil {
+		return kobject.KomposeObject{}, err
+	}
+
+	return komposeObject, nil
+}
+
+// Convert the Docker Compose v3 volumes to []string (the old way)
+// TODO: Check to see if it's a "bind" or "volume". Ignore for now.
+// TODO: Refactor it similar to loadV3Ports
+// See: https://docs.docker.com/compose/compose-file/#long-syntax-2
+func loadV3Volumes(volumes []types.ServiceVolumeConfig) []string {
+	var volArray []string
+	for _, vol := range volumes {
+
+		// There will *always* be Source when parsing
+		v := vol.Source
```

Comment:
docker-compose allows '_' in Volume names but Kubernetes does not.
`normalizeServiceNames` should be used here.